geode-benchmarks/src/main/java/org/apache/geode/benchmark/tests/PartitionedWithDeltaAndUniqueObjectReferenceBenchmark.java

```diff
@@ -0,0 +1,242 @@
+          .orElseThrow(() -> new IllegalStateException(
+              String.format("No Session with ID [%s] was found", id)));
+    }
+
+    private Session save(Session session) {
+      if (session != null && session.hasDelta()) {
+        this.region.put(session.getId(), session);
+        session.commit();
+      }
+
+      return session;
+    }
+
+    private int runSessionWorkload() {
+      return workloadTasks.stream().mapToInt(integerCallable -> {
+        try {
+          return integerCallable.call();
+        } catch (Exception e) {
+          e.printStackTrace();
+        }
+        return 0;
+      }).sum();
+    }
+
+    private Collection<Callable<Integer>> newSessionWorkloadTasks() {
+
+      Collection<Callable<Integer>> sessionWorkloadTasks = new LinkedList<>();
+
+      sessionWorkloadTasks.add(newAddSessionAttributeTask());
+      sessionWorkloadTasks.add(newRemoveSessionAttributeTask());
+      sessionWorkloadTasks.add(newSessionReaderTask());
+      return sessionWorkloadTasks;
+    }
+
+    private Callable<Integer> newAddSessionAttributeTask() {
+      return () -> {
+        Session session = findById(this.sessionId.get());
+
+        String name = UUID.randomUUID().toString();
+
+        session.setAttribute(name, System.currentTimeMillis());
+        save(session);
+
+        this.existingSessionAttributeNames.add(name);
+        return 1;
+      };
+    }
+
+    private Callable<Integer> newRemoveSessionAttributeTask() {
+      return () -> {
+        int returnValue = 0;
+
+        Session session = findById(this.sessionId.get());
+        String attributeName = null;
+
+        int size = this.existingSessionAttributeNames.size();
+        if (size > 0) {
+          int index = this.random.nextInt(size);
+          attributeName = this.existingSessionAttributeNames.remove(index);
+        }
+
+        if (session.getAttributeNames().contains(attributeName)) {
+          session.removeAttribute(attributeName);
+          returnValue = -1;
+        } else {
+          Optional.ofNullable(attributeName)
+              .filter(it -> !it.trim().isEmpty())
+              .ifPresent(this.existingSessionAttributeNames::add);
+        }
+
+        save(session);
+
+        return returnValue;
+      };
+    }
+
+    private Callable<Integer> newSessionReaderTask() {
+      return () -> {
+        Session session = findById(this.sessionId.get());
+        save(session.touch());
+
+        return 0;
+      };
+    }
+
+    private int safeFutureGet(Future<Integer> future) {
+      try {
+        return future.get();
+      } catch (Exception cause) {
+        throw new RuntimeException("Session access task failure", cause);
+      }
+    }
+  }
+
+  private class CreateClientPool implements Task {
+
+    @Override
+    public void run(TestContext context) throws Exception {
+      PoolFactory poolFactory = PoolManager.createFactory();
+      poolFactory
+          .addLocator("localhost", 10334)
```

Comment:
See other tests on how to get the locator and port. This falsely assumes the locator is localhost but when running on multiple hosts this may not be the case.